flink-state-backends/flink-statebackend-rocksdb/src/test/java/org/apache/flink/contrib/streaming/state/RocksDBWriteBatchWrapperTest.java

```diff
@@ -65,4 +69,52 @@ public void basicTest() throws Exception {
 			}
 		}
 	}
+
+	/**
+	 * Tests that {@link RocksDBWriteBatchWrapper} flushes after the memory consumed exceeds the preconfigured value.
+	 */
+	@Test
+	public void testWriteBatchWrapperFlushAfterMemorySizeExceed() throws Exception {
+		try (RocksDB db = RocksDB.open(folder.newFolder().getAbsolutePath());
+			WriteOptions options = new WriteOptions().setDisableWAL(true);
+			ColumnFamilyHandle handle = db.createColumnFamily(new ColumnFamilyDescriptor("test".getBytes()));
+			RocksDBWriteBatchWrapper writeBatchWrapper = new RocksDBWriteBatchWrapper(db, options, 200, 50)) {
+			// sequence (8 bytes) + count (4 bytes)
+			// more information please ref to write_batch.cc in RocksDB
+			assertEquals(12, writeBatchWrapper.getDataSize());
+			byte[] dummy = new byte[6];
+			ThreadLocalRandom.current().nextBytes(dummy);
+			// will add 1 + 1 + 1 + 6 + 1 + 6 = 16 bytes for each KV
+			// format is [handleType|kvType|keyLen|key|valueLen|value]
+			// more information please ref write_batch.cc in RocksDB
+			writeBatchWrapper.put(handle, dummy, dummy);
+			assertEquals(28, writeBatchWrapper.getDataSize());
+			writeBatchWrapper.put(handle, dummy, dummy);
+			assertEquals(44, writeBatchWrapper.getDataSize());
+			writeBatchWrapper.put(handle, dummy, dummy);
+			// will flush all, then an empty write batch
+			assertEquals(12, writeBatchWrapper.getDataSize());
+		}
+	}
+
+	/**
+	 * Tests that {@link RocksDBWriteBatchWrapper} flushes after the kv count exceeds the preconfigured value.
+	 */
+	@Test
+	public void testWriteBatchWrapperFlushAfterCountExceed() throws Exception {
+		try (RocksDB db = RocksDB.open(folder.newFolder().getAbsolutePath());
+			WriteOptions options = new WriteOptions().setDisableWAL(true);
+			ColumnFamilyHandle handle = db.createColumnFamily(new ColumnFamilyDescriptor("test".getBytes()));
+			RocksDBWriteBatchWrapper writeBatchWrapper = new RocksDBWriteBatchWrapper(db, options, 100, 50000)) {
+			byte[] dummy = new byte[2];
+			ThreadLocalRandom.current().nextBytes(dummy);
+			for (int i = 1; i < 100; ++i) {
+				writeBatchWrapper.put(handle, dummy, dummy);
+				// init 12 bytes, each kv consumes 8 bytes
+				assertEquals(12 + 8 * i, writeBatchWrapper.getDataSize());
```

Comment:
Ditto. Recording the init size of empty `writeBatchWrapper` instead using the hard-coded value `12`

Reply:
done